src/nlp/commands/dummy_data.py

```diff
@@ -0,0 +1,114 @@
+from nlp.load import import_main_class, prepare_module
+from nlp.utils import MockDownloadManager
+
+
+logger = logging.getLogger(__name__)
+
+
+def test_command_factory(args):
+    return DummyDataCommand(args.path_to_dataset, args.requires_manual,)
+
+
+class DummyDataCommand(BaseTransformersCLICommand):
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        test_parser = parser.add_parser("dummy_data")
+        test_parser.add_argument("--requires_manual", action="store_true", help="Dataset requires manual data")
+        test_parser.add_argument("path_to_dataset", type=str, help="Name of the dataset to download")
+        test_parser.set_defaults(func=test_command_factory)
+
+    def __init__(
+        self, path_to_dataset: str, requires_manual: bool,
+    ):
+        self._path_to_dataset = path_to_dataset
+        self._requires_manual = requires_manual
+        self._dataset_name = path_to_dataset.split("/")[-2]
+
+    def run(self):
+        module_path = prepare_module(self._path_to_dataset)
+        builder_cls = import_main_class(module_path)
+
+        # use `None` as config if no configs
+        configs = builder_cls.BUILDER_CONFIGS or [None]
+
+        for config in configs:
+            if config is None:
+                version = builder_cls.VERSION
+            else:
+                version = config.version
+
+            dataset_builder = builder_cls(config=config)
+            mock_dl_manager = MockDownloadManager(
+                dataset_name=self._dataset_name, config=config, version=version, is_local=True
+            )
+
+            dummy_data_folder = os.path.join(self._path_to_dataset, mock_dl_manager.dummy_data_folder)
+            logger.info(f"Creating dummy folder structure for {dummy_data_folder}... ")
+            os.makedirs(dummy_data_folder, exist_ok=True)
+
+            try:
+                generator_splits = dataset_builder._split_generators(mock_dl_manager)
+            except FileNotFoundError as e:
+
+                print(
+                    f"Dataset {self._dataset_name} with config {config} seems to already open files in the method `_split_generators(...)`. You might consider to instead only open files in the method `_generate_examples(...)` instead. If this is not possible the dummy data has to be created with less guidance. Make sure you create the file {e.filename}."
+                )
+
+            files_to_create = []
+            split_names = []
+            dummy_file_name = mock_dl_manager.dummy_file_name
+
+            for split in generator_splits:
+                logger.info(f"Collecting dummy data file paths to create for {split.name}")
+                split_names.append(split.name)
+                gen_kwargs = split.gen_kwargs
+                generator = dataset_builder._generate_examples(**gen_kwargs)
+
+                try:
+                    dummy_data_guidance_print = "\n" + 30 * "=" + "DUMMY DATA INSTRUCTIONS" + 30 * "=" + "\n"
```

Comment:
The idea is here to actually run the `_generate_examples` function and catch the `FileNotFoundError`. It will only catch the first `FileNotFoundError`, but most of the times there is only one `open(...)` statement in the function.